helper/hostutil/hostinfo_util.go

```diff
@@ -1,24 +1,51 @@
 package hostutil
 
+import (
+	"github.com/shirou/gopsutil/v3/host"
+	"github.com/shirou/gopsutil/v3/mem"
+)
+
 // VirutalMemoryStat holds commonly used memory measurements. We must have a
 // local type here in order to avoid building the gopsutil library on certain
 // arch types.
 type VirtualMemoryStat struct {
-	// Total amount of RAM on this system
-	Total uint64
+	*mem.VirtualMemoryStat
 
-	// RAM available for programs to allocate
+	// A subset of JSON struct tags in mem.VirtualMemoryStat were changed in a backwards
+	// incompatible way in the v3 release of gopsutil. The fields below are copied from
+	// v2 of gopsutil to maintain backwards compatibility in the Vault host-info API.
 	//
-	// This value is computed from the kernel specific values.
-	Available uint64
+	// The following details the changed JSON struct tags between v2 and v3:
+	// https://github.com/shirou/gopsutil/blob/master/_tools/v3migration/v3migration.sh#L61
+	CommitLimit    uint64 `json:"commitlimit"`
+	CommittedAS    uint64 `json:"committedas"`
+	HighFree       uint64 `json:"highfree"`
+	HighTotal      uint64 `json:"hightotal"`
+	HugePagesFree  uint64 `json:"hugepagesfree"`
+	HugePageSize   uint64 `json:"hugepagesize"`
+	HugePagesTotal uint64 `json:"hugepagestotal"`
+	LowFree        uint64 `json:"lowfree"`
+	LowTotal       uint64 `json:"lowtotal"`
+	PageTables     uint64 `json:"pagetables"`
+	SwapCached     uint64 `json:"swapcached"`
+	SwapFree       uint64 `json:"swapfree"`
+	SwapTotal      uint64 `json:"swaptotal"`
+	VMallocChunk   uint64 `json:"vmallocchunk"`
+	VMallocTotal   uint64 `json:"vmalloctotal"`
+	VMallocUsed    uint64 `json:"vmallocused"`
+	Writeback      uint64 `json:"writeback"`
+	WritebackTmp   uint64 `json:"writebacktmp"`
```

Comment:
JSON struct tags were changed for a subset of fields between gopsutil v2 and v3. I added these fields here to maintain backwards compatibility of the [host-info](https://www.vaultproject.io/api-docs/system/host-info) API response. For each of these fields, there is a duplicate field in the v3 `mem.VirtualMemoryStat` that has a different field name (e.g., `hugepagesfree` and `hugePagesFree`). This approach allows us to deprecate the v2 fields names and offer the v3 field names to transition clients over to.

Alternatively, we could map all of the fields in a v3 `mem.VirtualMemoryStat` into a struct that has the same JSON struct tags as v2. This would keep the host-info response identical to what it currently is, even though we'd be using v3 gopsutil structures internally. Looking for thoughts on the approach here.

Reply:
I think I like the alternative -- we have a struct for our API response, and we translate v3 gopsutil structures into that.

That way we could theoretically switch away from gopsutil to something else in the future, but still maintain our API response.

Reply:
I agree. I'll move forward with that 👍